Update limitations page to warn users that naming a gateway with the same name as a container causes an error (#1039)
radius-project/docs: docs/content/reference/limitations.md

See [app name constraints]({{< ref "resource-schema.md#common-values" >}}) for more information.

> ### Gateway resources and container resources cannot share names names

Author: ```suggestion
### Gateway resources and container resources cannot share names
```

Deploying a Radius Application that contains a gateway resource and a container resource that share a name will result in an error being thrown during the deployment process.

> 

Reviewer: Could you paste in ``` backticks the main string from the error? That way users can search in the docs and it will lead them to this page.

> As a workaround users can just name the respective resources differently.

Reviewer: ```suggestion
As a workaround make sure to use distinct names for both containers and gateways.
```

See [app name constraints]({{< ref "resource-schema.md#common-values" >}}) for more information.

### Gateway resources and container resources cannot share names

> Deploying a Radius Application that contains a gateway resource and a container resource that share a name will result in an error being thrown during the deployment process.

Reviewer: ```suggestion
Deploying a Radius Application that contains a gateway resource and a container resource that share a name will result in an error being thrown during deployment. For example, when attempting to name a container and a gateway something like "foo", you'll get an error messaging similar to:
```
